engine/src/main/java/org/entando/entando/aps/system/services/component/ComponentService.java

```diff
@@ -82,5 +80,67 @@ private List<Map<String, Object>> extractReferences(ComponentUsageDetails usage,
             throw new RestServerError("Error extracting references", e);
         }
     }
-
+    
+    private PagedMetadata<ComponentUsageEntity> extractUsageDetails(String componentCode, IComponentUsageService service) {
+        RestListRequest listRequest = new RestListRequest();
+        listRequest.setPageSize(0); // get all elements - no pagination
+        return service.getComponentUsageDetails(componentCode, listRequest);
+    }
+    
+    @Override
+    public ComponentDeleteResponse deleteInternalComponents(List<Map<String, String>> components) {
+        ComponentDeleteResponse response = new ComponentDeleteResponse();
+        log.debug("request deletion of components '{}'", components);
+        try {
+            response.setStatus(ComponentDeleteResponse.STATUS_SUCCESS);
+            components.stream().forEach(m -> {
+                String type = m.get(ComponentValidator.TYPE_FIELD);
+                String code = m.get(ComponentValidator.CODE_FIELD);
+                log.debug("Type '{}', Object code '{}'", type, code);
+                services.stream()
+                        .filter(s -> type.equalsIgnoreCase(s.getObjectType())).findFirst()
+                        .ifPresent(service -> {
+                            Map<String, Object> properties = new HashMap<>();
+                            properties.put(REFERENCE_TYPE_PROPERTY, type);
+                            properties.put(REFERENCE_CODE_PROPERTY, code);
+                            try {
+                                service.getComponentDto(code).ifPresentOrElse(dto -> {
+                                    PagedMetadata<ComponentUsageEntity> result = this.extractUsageDetails(code, service);
+                                    if (this.checkReferences(components, result.getBody())) {
+                                        service.deleteComponent(code);
```

Comment:
@mirkoPan I have found another big problem (Warning!!!) here: 
- if one delete goes in error it will throw an exception and it will exit but it should just set the current status (error for the row and partial success or FAILURE for the global status) without exit
- A "failure" for the global status is never used: if all deletes go in error you should set failure!

Reply:
done

Reply:
done